CMakeLists.txt

```diff
@@ -83,6 +98,8 @@ if (CMAKE_VERSION VERSION_LESS 3.1 OR NOT USE_COMPILE_FEATURES)
   endif ()
 endif ()
 
+set (CMAKE_CXX_FLAGS "-O3 -g -march=native ${CMAKE_CXX_FLAGS}")
```

Comment:
This is unrelated to tbb.

Reply:
> This is unrelated to tbb.

It is not, but I had branched from the master that this existed. If you remember we had a discussion about this (#161). Note that this way is the proper way of adding the flags (`-march=native`). We could merge them both and not have different PRs.

Reply:
I dont agree, this unrelated and should be part of another PR.
This is not proper at all, someone would want to set completely different flags depending on the use case: I think these should not be touched.
Plus their're not accepted by all compilers.

Reply:
> I dont agree, this unrelated and should be part of another PR.

Will make another PR.

> This is not proper at all, someone would want to set completely different flags depending on the use case

Let's make the library header only then. It's the only way of letting the user set the different flags that they want.

> Plus their're not accepted by all compilers.

`-march=native` is accepted by GCC and Clang. I can make a version for ICC (which has `-mtune=native`). In the new PR that I will make, you could test also for Visual Studio (although AppVeyor does not complain at all).

Reply:
No need for header only: leave cmake defaults so that the user can set CMAKE_CXX_FLAGS to whatever he wants. A lot of compiler exist that are different from gcc/clang. Old versions of msvc accept unknown flags but emit a warning. I think more recent versions reject them.

Reply:
> No need for header only: leave cmake defaults so that the user can set CMAKE_CXX_FLAGS to whatever he wants. A lot of compiler exist that are different from gcc/clang. Old versions of msvc accept unknown flags but emit a warning. I think more recent versions reject them.

I removed everything related to optimized flags from this PR. Will leave it for another and we can continue this discussion there..